Release initial Auto SDK docs

EmilFattakhov opened the pull request:
This PR consists of two parts:
- Initial release of the Auto SDK documentation
- Rework of the developer documentation structure in order to accommodate Auto SDK and explain the difference between Auto SDK and Nova

jim-counter commented on docs/develop/intro.md:
```diff
@@ -0,0 +1,48 @@
+
+Absolutely not! Our primary objective is to minimize barriers to entry for both farmers and developers.
+
+We pride ourselves on offering unlimited possibilities—there are no boundaries!
```
```suggestion
We pride ourselves on enabling unlimited possibilities - there are no boundaries!
```

jim-counter commented on docs/develop/intro.md:
```diff
@@ -0,0 +1,48 @@
+## [Auto SDK](/docs/develop/auto_sdk/intro)
+
+### Key Features:
+- **Modular Packages**: The SDK is organized into packages like auto-utils, auto-consensus, and auto-id, allowing you to use only what you need.
```
```suggestion
- **Modular Packages**: The SDK is organized into packages like auto-utils, auto-consensus, and auto-id. This partitions functionality intuitively and allows you to import only what you need.
```

jim-counter commented on docs/develop/intro.md:
```diff
@@ -0,0 +1,48 @@
+
+### Key Features:
+- **Solidity Smart Contracts**: Write and deploy smart contracts using Solidity, just as you would on Ethereum.
+- **Tool Compatibility**: Any tool available for Ethereum development—like Remix, Truffle, or Hardhat—is compatible with Autonomys.
```
```suggestion
- **Tool Compatibility**: Any tool available for Ethereum development - like Remix, Truffle, or Hardhat - is compatible with Nova.
```

jim-counter commented on docs/develop/auto_sdk/intro.md:
```diff
@@ -0,0 +1,124 @@
+---
+
+## Introduction to the Auto SDK
+Welcome to the Auto SDK documentation! The Auto SDK is a comprehensive toolkit designed to simplify the development of applications and integrations with the Autonomys Network. It provides a set of modular packages that enable developers to interact seamlessly with the consensus chains and active domains (Auto ID, Nova)
```
```suggestion
Welcome to the Auto SDK documentation! The Auto SDK is a comprehensive toolkit designed to simplify the development of applications and integrations with the Autonomys Network. It provides a set of modular packages that enable developers to interact seamlessly with the consensus chain and active domains (Auto ID, Nova)
```

jim-counter commented on docs/develop/auto_sdk/intro.md:
```diff
@@ -0,0 +1,124 @@
+Welcome to the Auto SDK documentation! The Auto SDK is a comprehensive toolkit designed to simplify the development of applications and integrations with the Autonomys Network. It provides a set of modular packages that enable developers to interact seamlessly with the consensus chains and active domains (Auto ID, Nova)
+
+### What is the Auto SDK?
+The Auto SDK is a collection of JavaScript/TypeScript packages that abstract the complexity of blockchain interactions. It includes utilities for consensus mechanisms, identity management, and general-purpose functions that are essential for building decentralized applications (dApps) and services on the Autonomys Network.
```
```suggestion
The Auto SDK is a collection of JavaScript/TypeScript packages that abstract away the complexity of blockchain interactions. It includes utilities for consensus interactions, identity management, and general-purpose functions that are essential for building decentralized applications (dApps) and services on the Autonomys Network.
```

jim-counter commented on docs/develop/auto_sdk/intro.md:
```diff
@@ -0,0 +1,124 @@
+### Repository Structure
+The repository is organized as follows:
+
+- packages/auto-utils: Utility functions for the SDK.
```
```suggestion
- `packages/auto-utils`: Utility functions for the SDK.
```

jim-counter commented on docs/develop/auto_sdk/intro.md:
```diff
@@ -0,0 +1,124 @@
+The repository is organized as follows:
+
+- packages/auto-utils: Utility functions for the SDK.
+- packages/auto-consensus: Consensus-related functions.
```
```suggestion
- `packages/auto-consensus`: Consensus-related functions.
```

jim-counter commented on docs/develop/auto_sdk/intro.md:
```diff
@@ -0,0 +1,124 @@
+
+- packages/auto-utils: Utility functions for the SDK.
+- packages/auto-consensus: Consensus-related functions.
+- packages/auto-id: Identity management functions.
```
```suggestion
- `packages/auto-id`: Identity management functions.
```

jim-counter commented on docs/develop/auto_sdk/intro.md:
```diff
@@ -0,0 +1,124 @@
+You can test the SDK packages against a local Autonomys node using the provided scripts instead of a public testnet (**Gemini 3h**).
+
+**Steps**:
+1. Verify OS and Architecture Settings
```
This and the next line are run together. May need a `<br />` or similar. Maybe bold the step name too?

jim-counter commented on docs/develop/auto_sdk/intro.md:
````diff
@@ -0,0 +1,124 @@
+ARCHITECTURE="aarch64" # Options: aarch64 | x86_64-skylake | x86_64-v2
+```
+
+2. Run the Development Script
````
This and the next line are run together. May need a `<br />` or similar. Maybe bold the step name too?

jim-counter reviewed (changes requested):
We mention `api: Api` in a number of method signatures on the auto-consensus page but do not provide any insight into what the object is. Would be nice to work a definition in somewhere if possible.

I wonder if the lists of methods would be better as a two column table with method and description as the columns? And could we sort them alphabetically please?

Reads great though I have not followed the process to test.

ryyeung commented on docs/develop/auto_sdk/auto-xdm.md:
```diff
@@ -0,0 +1,43 @@
+
+- `transfer(api: ApiPromise, destination: ChainOrDomain, receiver: string, amount: Amount): Promise<SubmittableExtrinsic>`: Creates a transaction to transfer a specified amount to a receiver on a given destination, which can be either the consensus chain or a specific domain.
+
+#### Types:
```
The "Types" sizing looks abnormally small here. Perhaps ### would be a better header size.

ryyeung commented on docs/develop/auto_sdk/auto-xdm.md:
```diff
@@ -0,0 +1,43 @@
+
+---
+
+**Usage Example:**
```
Larger header would be good here imo. The usage example is what readers will look for first.

ryyeung commented on docs/develop/auto_sdk/auto-id.md:
```diff
@@ -0,0 +1,68 @@
+---
+title: Auto ID
```
Do we have any usage examples for AutoID functions?

EmilFattakhov replied:
I can add a few, np 👍

ryyeung commented on docs/develop/auto_sdk/auto-consensus.md:
```diff
@@ -0,0 +1,88 @@
+---
+title: Auto Consensus
```
Would providing usage examples be good here?

ryyeung reviewed (changes requested):
The documents look really good Emil! Everything was formatted well when I ran it on my local machine.

Usage examples would be great to add here, if possible!

EmilFattakhov commented:
Okay, most of the comments were addressed. 
@jim-counter 
1. I think it's a good idea to provide some more information about the API, but the return object is so massive that it's almost indescribable. Let me think about what information can be added, but hopefully, the **Usage Examples** will make things slightly easier. 
2. I liked the idea of the tables, but when implemented I faced a few issues:
- Tables have inconsistent width because the functions and their descriptions differ a lot in length. It doesn't look good in my opinion. 
![Screenshot 2024-10-02 at 1 19 39 PM](https://github.com/user-attachments/assets/d0ce16ad-d516-41af-beb9-c9363e230a50)
![Screenshot 2024-10-02 at 1 21 57 PM](https://github.com/user-attachments/assets/5793265a-88bc-4435-98a5-e31d989fd52c)

- But secondly and most importantly: the .mdx file becomes almost unreadable because the functions have a lot of complex syntax you have to escape a lot of characters (like ```, `{` and` |`)  and it gets very difficult to maintain the file, so I decided to skip adding the tables for the moment. 
3. All functions and modules have been sorted alphabetically, added a usage example and will likely add some more. 
4. Added the information about RPCs

jim-counter reviewed (approved):
LGTM! 🚀
